### PR TITLE
OnDbNuke hook

### PR DESCRIPTION
**Pull request by joshblum**
Adds `DbNukeHooks` to `GlobalContext` to unify clearing of caches.

### CODE REVIEW

File: go/stellar/global.go [hunk 1]
```diff
@@ -27,7 +27,9 @@ func ServiceInit(g *libkb.GlobalContext, walletState *WalletState, badger *badge
 	if g.Env.GetRunMode() != libkb.ProductionRunMode {
 		stellarnet.SetClientAndNetwork(horizon.DefaultTestNetClient, build.TestNetwork)
 	}
-	g.SetStellar(NewStellar(g, walletState, badger))
+	s := NewStellar(g, walletState, badger)
+	g.SetStellar(s)
+	g.AddLogoutHook(s, "stellar")
```
Comment (joshblum) [inline go/stellar/global.go]: cc @mlsteele @patrickxb is there anything we want to clear in stellar on a dbnuke? from a quick pass I didn't see anything obvious

**Comment by joshblum** [comment]
nope that happens separately on logout

File: go/stellar/global.go (hunk 1 above)
Comment (patrickxb) [reply go/stellar/global.go]: stellar doesn't use leveldb.

Comment (joshblum) [reply go/stellar/global.go]: @patrickxb goal is to have the dbnuke hook clear any disk/memory caches not just leveldb

Comment (patrickxb) [reply go/stellar/global.go]: then you would want to do what happens in service/main.go:OnLogout:

            log("resetting wallet state on logout")
            if d.walletState != nil {
                    d.walletState.Reset(m)                                                    
            }

on that Service.walletState variable.

**Pull request merged**